ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -203,23 +218,91 @@ func (h *FaultHandler) CheckNetworkBlackHolePort() func(http.ResponseWriter, *ht
 		rwMu.RLock()
 		defer rwMu.RUnlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Return the correct status state
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("running")
-		logger.Info("Successfully checked status for fault", logger.Fields{
-			field.RequestType: requestType,
-			field.Request:     request.ToString(),
-			field.Response:    responseBody.ToString(),
-		})
+		ctx := context.Background()
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, requestTimeoutDuration)
+		defer cancel()
+
+		var responseBody types.NetworkFaultInjectionResponse
+		var statusCode int
+		chainName := fmt.Sprintf("%s-%s-%s", *request.TrafficType, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+		running, cmdOutput, cmdErr := h.checkNetworkBlackHolePort(ctxWithTimeout, request, chainName, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+
+		// We've timed out trying to check if the black hole port fault injection is running
+		if err := ctx.Err(); err != nil {
+			logger.Error("Request timed out", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Error:       err,
+			})
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(requestTimedOutError)
+		} else if cmdErr != nil {
+			logger.Error("Unknown error encountered for request", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Error:       cmdErr,
```

Comment:
non-blocking - can we log the cmdOutput?

Reply:
we're already logging this on L.294

Reply:
I see. Thanks.